--- a/Thirdparty/g2o/CMakeLists.txt
--- b/Thirdparty/g2o/CMakeLists.txt
@@ -67,9 +67,11 @@
 
 # Find Eigen3
 SET(EIGEN3_INCLUDE_DIR ${G2O_EIGEN3_INCLUDE})
-FIND_PACKAGE(Eigen3 3.1.0 REQUIRED)
+#FIND_PACKAGE(Eigen3 3.1.0 REQUIRED)
+FIND_PACKAGE(PkgConfig)
+pkg_check_modules( EIGEN3 REQUIRED eigen3>=3.1.0 )
 IF(EIGEN3_FOUND)
-  SET(G2O_EIGEN3_INCLUDE ${EIGEN3_INCLUDE_DIR} CACHE PATH "Directory of Eigen3")
+  SET(G2O_EIGEN3_INCLUDE ${EIGEN3_INCLUDE_DIRS} CACHE PATH "Directory of Eigen3")
 ELSE(EIGEN3_FOUND)
   SET(G2O_EIGEN3_INCLUDE "" CACHE PATH "Directory of Eigen3")
 ENDIF(EIGEN3_FOUND)
@@ -172,3 +174,4 @@
 g2o/stuff/property.cpp
 g2o/stuff/property.h
 )
+INSTALL(TARGETS g2o DESTINATION lib)
