--- a/CMakeLists.txt		
+++ b/CMakeLists.txt.g2o	
@@ -248,10 +248,12 @@
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${g2o_C_FLAGS}")
 
 # Find Eigen3
-FIND_PACKAGE(Eigen3 REQUIRED)
+# FIND_PACKAGE(Eigen3 REQUIRED)
+FIND_PACKAGE(PkgConfig)
+pkg_check_modules(EIGEN3 REQUIRED eigen3)
 
 # Set up the top-level include directories
-INCLUDE_DIRECTORIES(${g2o_SOURCE_DIR} ${EIGEN3_INCLUDE_DIR})
+INCLUDE_DIRECTORIES(${g2o_SOURCE_DIR} ${EIGEN3_INCLUDE_DIRS})
 
 # Generate config.h
 SET(G2O_OPENGL_FOUND ${OPENGL_FOUND})
